flaml/tune/tune.py

```diff
@@ -504,15 +503,19 @@ def easy_objective(config):
             try:
                 import optuna as _
 
-                SearchAlgorithm = BlendSearch
+                SearchAlgorithm = BlendSearch if not search_alg else eval(search_alg)

```

Comment:
The use of `eval` is not safe. Could you use an alternative way?

Reply:
I add assertion to make it feasible.

Reply:
First, I don't like the use of `eval`. Unless you get an approval from an experience Python engineer, I'm against it. Even `locals` is better.
Second, the code doesn't look correct in the try...except block. The try part tries to import optuna before using BlendSearch. If the choice is not BlendSearch, we shouldn't enter the try...except block.